### PR TITLE
perf(kimi-k3): tune small-M MoE decode

### PR DESCRIPTION
**Pull request by panditsa**
Summary

- extend the joint latent/shared decode pipeline from M=1 through M=4
- rebalance routing and W2 work for M=2/4
- tune TP-local A16W4 tiles and split top-k route work across CTAs

Performance

4K/1K TP8/EP8 CUDA graphs, median of three:

| M | Prior equivalent stack tok/s | Head tok/s | Change | Prior TPOT | Head TPOT |
|---:|---:|---:|---:|---:|---:|
| 2 | 84.77 | 93.39 | +10.2% | 22.864 ms | 20.663 ms |
| 4 | 152.23 | 156.34 | +2.70% | 24.954 ms | 24.255 ms |

The comparison uses the last equivalent EP8 boundary: intervening #1137/#1138 only add unselected A8W4 foundations and #1139 registers the exact TP8/EP1 shape. The measured EP8 execution path is unchanged until this PR.

Tests

- BF16 SiTU M=1/2/3/4 coverage: 4 passed
- router dispatch/per-token FP32 route coverage: 2 passed
- fused RMSNorm + projection M=1/2/3/4 graph coverage: 4 passed
- all 18 measured 4K/1K requests completed with valid lengths

Stack

- depends on #1139

### CODE REVIEW

**Comment by jhinpan** [comment]
Exact #1132 screen of the only M=8-applicable change here (the sigmoid-router warp-count line), applied independently to current main (`9196fa27`) with forced acceptance=1 at 131072/1024 TP8/EP8:

- baseline: 36.80 ms TPOT, 19.7647 tok/s
- router candidate: 36.77 ms TPOT, 19.7753 tok/s

That is only ~+0.05% throughput / 0.03 ms and is not an issue-level material win. The larger W13/W2/pipeline changes remain gated to M<=4 or the TP-local EP1 shape, so I will not extract this line into a separate #1132 PR.

File: python/tokenspeed/runtime/models/kimi_k3.py [hunk 1]
```diff
@@ -1821,7 +1821,7 @@ def forward(
             max_num_tokens_per_gpu = num_global_tokens
 
         if self.native_latent_moe is not None:
-            if self._use_fused_decode_pipeline and hidden_states.shape[0] == 1:
+            if self._use_fused_decode_pipeline and hidden_states.shape[0] <= 4:
```
[the remark this answers is not in the document]
Comment (panditsa) [reply python/tokenspeed/runtime/models/kimi_k3.py]: Addressed in ce0a5409. The fused latent-decode path now requires 0 < M <= 4, so zero-token forwards continue through the native empty-safe path. Added a focused zero-row runtime regression.

File: tokenspeed-kernel/python/tokenspeed_kernel/ops/moe/gluon/sigmoid_topk.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def gluon_sigmoid_bias_topk_gfx950(
     ) -> tuple[torch.Tensor, torch.Tensor]:
         route = (
             invoke_sigmoid_bias_topk_route_gluon
-            if router_logits.shape[0] * topk <= 128
+            if router_logits.shape[0] == 1 or router_logits.dtype != torch.float32
             else invoke_sigmoid_bias_topk_route_prefill_gluon
```
[the remark this answers is not in the document]
Comment (panditsa) [reply tokenspeed-kernel/python/tokenspeed_kernel/ops/moe/gluon/sigmoid_topk.py]: Addressed in ce0a5409. FP32 routing now uses the prefill launcher only when topk <= 16; larger top-k values retain the decode launcher. Added a forced-Gluon M=2/top-k=32 dispatch regression.
